scripts/leaderboard.coffee

```diff
@@ -116,23 +117,31 @@ module.exports = (robot) ->
     # for each ++/--
     for i in [0...msg.match.length]
       testword = msg.match[i]
-
-      # updates Scoring for words, accordingly and returns result string
-      result = updateScore(testword, ScoreField, msg.message.user.name)
-
-
-      end = start + testword.length
-
-      # generates response message for reply
-      if result.Response == "-1"
-        newmsg = "#{testword} [Sorry, You can't give ++ or -- to yourself.]"
+      
+      # search for user
+      util.info (body) ->
+      result = parse body, testword.toLowerCase()
+      
+      if not result
+        continue
       else
-        newmsg = "#{testword} [#{result.Response} #{result.Name} now at #{result.New}] "
-      oldmsg = oldmsg.substr(0, start) + newmsg + oldmsg.substr(end+1)
-      start += newmsg.length
-
-    # reply with updated message
-    msg.send "#{oldmsg}"
+        # get username from search result
+        username = result[10]
+        # update score for user (slack ID)
+        result = updateScore(username, ScoreField, msg.message.user.name)
+        # update search index
+        end = start + testword.length
+        # generate response message for reply
+        if result.Response == "-1"
+          newmsg = "#{testword} [Sorry, You can't give ++ or -- to yourself.]"
+        else
+          newmsg = "#{testword} [#{result.Response} #{result.Name} now at #{result.New}] "
+        oldmsg = oldmsg.substr(0, start) + newmsg + oldmsg.substr(end+1)
+        start += newmsg.length
+    
+    # reply with updated message, if modified
+    if oldmsg != msg.message.text
```

Comment:
Why have you added this check?

Reply:
[This loop](https://github.com/BURG3R5/bot/blob/master/scripts/leaderboard.coffee#L118) iterates over every `++`/`--`. [This if](https://github.com/BURG3R5/bot/blob/master/scripts/leaderboard.coffee#L121) checks if the `++`/`--` is valid. The means only the first `++` in "@bert++ for mastering C++" will be operated on and modified in the reply message. Now, if _all_ the `++`/`--` in the message are invalid, like "C++ is a nightmare!", [this if](https://github.com/BURG3R5/bot/blob/master/scripts/leaderboard.coffee#L143) prevents an unmodified message from being sent.

In other words, if the message is not changed by Bro, don't send it.

Reply:
Cool